Add .github/stale.yaml for torch.distributed issues.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60038 Add .github/stale.yaml for torch.distributed issues.**

Followed the instructions in  https://github.com/probot/stale to
enable the probot-stale github bot to automatically purge stale issues.

Enabled it only for the label "oncall: distributed" for now.

The goal is to automatically mark and purge stale issues which is a good forcing function for users and developers to follow up on an issue or close it out because it is not actionable.

Differential Revision: [D29144396](https://our.internmc.facebook.com/intern/diff/D29144396/)

pritamdamania committed:
Add .github/stale.yaml for torch.distributed issues.

Followed the instructions in  https://github.com/probot/stale to
enable the probot-stale github bot to automatically purge stale issues.

Enabled it only for the label "oncall: distributed" for now.

Differential Revision: [D29144396](https://our.internmc.facebook.com/intern/diff/D29144396/)

[ghstack-poisoned]

pritamdamania87 commented:
@ezyang I did add probot-stale to the pytorch account, but I don't think I have permissions to view the pytorch organization's settings, so not sure if it is really enabled.

pritamdamania committed:
Update on "Add .github/stale.yaml for torch.distributed issues."


Followed the instructions in  https://github.com/probot/stale to
enable the probot-stale github bot to automatically purge stale issues.

Enabled it only for the label "oncall: distributed" for now.

The goal is to automatically mark and purge stale issues which is a good forcing function for users and developers to follow up on an issue or close it out because it is not actionable.

Differential Revision: [D29144396](https://our.internmc.facebook.com/intern/diff/D29144396/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Add .github/stale.yaml for torch.distributed issues."


Followed the instructions in  https://github.com/probot/stale to
enable the probot-stale github bot to automatically purge stale issues.

Enabled it only for the label "oncall: distributed" for now.

The goal is to automatically mark and purge stale issues which is a good forcing function for users and developers to follow up on an issue or close it out because it is not actionable.

Differential Revision: [D29144396](https://our.internmc.facebook.com/intern/diff/D29144396/)

[ghstack-poisoned]

cbalioglu commented:
The linter complains about trailing spaces; otherwise looks good to me. Let's hear from @ezyang whether we can start using it.

pritamdamania committed:
Update on "Add .github/stale.yaml for torch.distributed issues."


Followed the instructions in  https://github.com/probot/stale to
enable the probot-stale github bot to automatically purge stale issues.

Enabled it only for the label "oncall: distributed" for now.

The goal is to automatically mark and purge stale issues which is a good forcing function for users and developers to follow up on an issue or close it out because it is not actionable.

Differential Revision: [D29144396](https://our.internmc.facebook.com/intern/diff/D29144396/)

[ghstack-poisoned]

ezyang commented:
I generally have a dim view of using stale bots, period. https://blog.benwinding.com/github-stale-bots/ does a reasonable job of explaining my perspective. If you want to stale bot flaky test issues, that's another matter, but bug reports posted by real, breathing users? Treat them like gold dust.

pritamdamania87 commented:
> I generally have a dim view of using stale bots, period. https://blog.benwinding.com/github-stale-bots/ does a reasonable job of explaining my perspective.

The blog seems to be pointing at the problem in angular where the issue is locked where the user can no longer update the issue. If that is the major concern, my understanding is that this bot will just mark it stale in X days and close it after Y days, but doesn't lock the issue at all. Users can re-open the issue and continue to add more context afterwards as well.

Also, I think we can have this bot exclude certain labels, for example for certain issues we could tag them with a label like "no-probot-stale" and the bot would ignore them.

ezyang commented:
Let me go back to the goal:

> The goal is to automatically mark and purge stale issues which is a good forcing function for users and developers to follow up on an issue or close it out because it is not actionable.

Separating the goals out, you are looking for:

* A way to get users to follow up on issues after you've asked them for more information
* A way to close non actionable issues

A bot is a reasonable way to send a reminder N days later for a user to give us more information, and even close it if no information is forthcoming. I'd examine, however, why you want to *close* so called "non actionable" issues. Is it because, without closing non actionable issues, there are too many issues?

Closing issues because you have "too many issues" is the wrong way to go about doing it. It's saying "we have a huge backlog because [our bug ingress is greater than our bug egress](https://apenwarr.ca/log/20171213#slide29), therefore let's just close all our old issues." You haven't solved the fundamental problem, which is that you have more bugs than time to fix them all (let's face it, what popular project isn't in this situation), and you've also managed to make the people who filed these bugs in the first place [angry](https://www.reddit.com/r/programming/comments/kzvryq/github_stale_bots_a_false_economy/) (because no one wants to see the bug report they laboriously wrote out get unceremoniously closed by a bot months later.) If this really is your problem, I suggest using priorities (e.g., the [high priority](https://github.com/pytorch/pytorch/issues?q=is%3Aopen+is%3Aissue+label%3A%22high+priority%22) label, as used in the main triage oncall) to talk about what bugs you actually are prioritizing to fix, and which ones you just aren't going to do (this cycle, anyway). pytorch/pytorch's issue count keeps growing, but we've done a pretty good job of keeping the high priority issue count stable.

Yes, sometimes an issue really is just old and it's better to just close it and rereport if it happens again. But this happens on the scale of years if the component in question doesn't change very much. I very much agree with apenwarr's advice that you should just periodically retriage old issues and have a human decide if something really is old or not. Having the system fail "closed" (as in, if there is no intervention the issue gets closed) is the wrong way of going about it, because there isn't really any reason to believe a old bug got fixed if, you know, you haven't done anything that would have fixed it (only a human can know that).

pritamdamania87 commented:
I don't have strong opinions on this and I'm fine with the proposal @ezyang suggested where we continue our practice of tagging hi-pri issues and have goals around addressing those and don't have specific goals around addressing all issues in our backlog.

cc @cbalioglu @mrshenli Any further thoughts on this?
